### ISSUE TITLE
[v4] Replace `SFC` with `FunctionComponent`

### ISSUE THREAD

**Issue by imhele** [ant-design/ant-design-pro#3725]
```ts
/**
 * @deprecated as of recent React versions, function components can no
 * longer be considered 'stateless'. Please use `FunctionComponent` instead.
 *
 * @see [React Hooks](https://reactjs.org/docs/hooks-intro.html)
 */
type SFC<P = {}> = FunctionComponent<P>;
```

**Comment by xiaohuoni** [comment]
好像SFC跳转到定义的时候，就是FunctionComponent

**Comment by imhele** [comment]
Closed via #3723
